addon/components/ya-input/component.js

```diff
@@ -3,89 +3,86 @@ import layout from '../../templates/ya-input';
-  canShowErrors: false,
+  focusedOnce: false,
   hasLostFocus: false,
+  initialFocus: false,
 
-  errorText: oneWay('errors.firstObject'),
+  model: oneWay('form.model'),
+  modelName: oneWay('model.constructor.modelName'),
+  shouldShowValidationErrors: oneWay('model.shouldShowValidationErrors'),
 
-  fieldObserver: observer('value', function() {
-    const props  = getProperties(this, 'form.model', 'field-name');
-    set(props['form.model'], props['field-name'], get(this, 'value'));
+  errorText: computed('errors.[]', {
+    get() {
+      return get(this, 'errors').join(', ');
+    },
+    set: defaultCPSetter
   }),
 
-  setErrorsObserver: on('init', observer('form.model', 'field-name', function() {
-    const fieldName = get(this, 'field-name');
-    const valueKey = `form.model.${fieldName}`;
-    const errorsKey = `form.model.errors.${fieldName}`;
+  inputId: computed('model', 'field-name', {
+    get() {
+      const modelName = getWithDefault(this, 'modelName', get(this, 'model').toString());
 
-    if (isEmpty(fieldName)) {
-      // might want to teardown
-      return;
-    }
+      return dasherize(`${modelName}-${get(this, 'field-name')}`);
+    },
+    set: defaultCPSetter
+  }),
 
-    addObserver(this, valueKey, this, this._updateValue);
-    addObserver(this, errorsKey, this, this._updateErrors);
+  name: computed('fieldName', {
+    get() {
+      return dasherize(get(this, 'field-name'));
+    },
+    set: defaultCPSetter
+  }),
 
-    this._updateValue(this, valueKey);
-    this._updateErrors(this, errorsKey);
-  })),
+  value: computed('model', 'field-name', {
+    get() {
+      const model = get(this, 'model');
+      const fieldName = get(this, 'field-name');
 
-  _updateValue(component, key) {
-    set(component, 'value', get(component, key));
-  },
+      return model ? get(model, fieldName) : null;
+    },
+    set(_key, newValue) {
+      const model = get(this, 'model');
+      const fieldName = get(this, 'field-name');
 
-  _updateErrors(component, key) {
-    set(component, 'errors', get(component, key));
-  },
+      return model ? set(model, fieldName, newValue) : null;
+    }
+  }),
 
-  errorsObserver: observer('errors.length', 'hasLostFocus', 'form.model.shouldShowValidationErrors', function() {
-    runNext(() => {
+  canShowErrors: computed('errors.[]', 'hasLostFocus', 'shouldShowValidationErrors', {
+    get() {
       const errorsCount = get(this, 'errors.length');
       const hasLostFocus = get(this, 'hasLostFocus');
-      const shouldShowValidationErrors = get(this, 'form.model.shouldShowValidationErrors');
+      const shouldShowValidationErrors = get(this, 'shouldShowValidationErrors');
 
-      if (errorsCount === 0) {
-        set(this, 'canShowErrors', false);
-      } else if (errorsCount > 0 && (hasLostFocus || shouldShowValidationErrors)) {
-        set(this, 'canShowErrors', true);
-      }
-    });
+      return !!(errorsCount > 0 && (hasLostFocus || shouldShowValidationErrors));
```

Comment:
I know this `shouldShowValidationErrors` already existed, but I find it pretty confusing to have a function named `canShowErrors` and another one called `shouldShowValidationErrors` – nevermind having one called within the other.

One of these functions should be renamed to more accurately describe what it's doing.